### PR TITLE
feat: add eslint mocha plugin and rule per issue #6

### PR DESCRIPTION
**Pull request by morrme**
Resolves issue #6

### CODE REVIEW

**Comment by morrme** [comment]
Other rules are located at https://github.com/lo1tuma/eslint-plugin-mocha/tree/master/docs/rules .

File: package.json [hunk 1]
```diff
@@ -10,7 +10,8 @@
   },
   "devDependencies": {
     "eslint": "3",
-    "mocha": "3"
+    "mocha": "3",
+    "eslint-plugin-mocha": "^4.9.0"
```
Comment (vladikoff) [inline package.json]: lock this to `4.9.0` (remove the `^`)

File: lib/configs/default.js [hunk 2]
```diff
@@ -11,6 +11,11 @@ module.exports = {
     es6: true, // enable all ECMAScript 6 features except for modules
     node: true // Node.js global variables and Node.js-specific rules
   },
+  
+  plugins: [
+        "mocha"
```
Comment (vladikoff) [inline lib/configs/default.js]: formatting is off here

**Comment by vladikoff** [comment]
good start! ~~shrinkwrap needs to regenerated to add this new devDep into the source. run `npm run shrinkwrap`~~

**Comment by vladikoff** [comment]
Nevermind, ignore comment above about shrinkwrap, wrong repository!

File: lib/configs/default.js (hunk 2 above)
Comment (morrme) [reply lib/configs/default.js]: @vladikoff aside from moving it over, should i go to single quotes as well?

File: package.json [hunk 3]
```diff
@@ -10,7 +10,8 @@
   },
   "devDependencies": {
     "eslint": "3",
-    "mocha": "3"
+    "mocha": "3",
+    "eslint-plugin-mocha": "4.9.0"
```
Comment (seanmonstar) [inline package.json]: I *think* that with this just being a dev dependency, it won't actually be included when other repos use eslint-plugin-fxa. A way to be sure is if you adjust the dependency in the fxa-auth-server to point at this branch, npm update, add `only` to one of the tests, and then run `grunt eslint`, it should fail the lint.

Comment (morrme) [reply package.json]: @seanmonstar i am working on testing now. didn't work on one of my machines, so i must try it on another....

Comment (morrme) [inline package.json]: @seanmonstar ok ` grunt eslint` passed, so your theory may be correct. i will now try the other way.....

Comment (morrme) [reply package.json]: @seanmonstar ok did some more testing and it actually works both in `dependencies` and `devDependencies`.  The fxa-auth-server test failed with both of them.

So given that it works both ways, where would you like the dependency to be?

Comment (vladikoff) [reply package.json]: @morrme here's what I get:

```
➜  fxa-auth-server git:(master) npm i morrme/eslint-plugin-fxa#add-mocha-plugin
eslint-plugin-fxa@1.0.0 node_modules/eslint-plugin-fxa
➜  fxa-auth-server git:(master) grunt eslint
Running "eslint:files" (eslint) task
Warning: Failed to load plugin mocha: Cannot find module 'eslint-plugin-mocha'
Referenced from: plugin:fxa/server
Referenced from: /Users/vladikoff/mozilla/fxa-local-dev/fxa-auth-server/.eslintrc Use --force to continue.
```
How did you test it?

Comment (morrme) [reply package.json]: @vladikoff I got that message too so I installed the plugin, then tested from there.

Comment (vladikoff) [reply package.json]: @morrme ah. when we install via `npm i morrme/eslint-plugin-fxa#add-mocha-plugin` it should also install the dependency module

Comment (morrme) [reply package.json]: @vladikoff Ah, so would moving it to `dependencies` actually fix that?

Comment (vladikoff) [reply package.json]: @morrme I think it should

**Comment by vladikoff** [comment]
@morrme I still get the same issue if I do: 

```
➜  fxa-auth-server git:(master) npm i morrme/eslint-plugin-fxa#add-mocha-plugin
npm WARN peerDependencies The peer dependency eslint@^2.0.0 || ^3.0.0 included from eslint-plugin-mocha will no
npm WARN peerDependencies longer be automatically installed to fulfill the peerDependency 
npm WARN peerDependencies in npm 3+. Your application will need to depend on it explicitly.
eslint-plugin-fxa@1.0.0 node_modules/eslint-plugin-fxa
├── eslint@3.19.0 (path-is-inside@1.0.2, natural-compare@1.4.0, strip-bom@3.0.0, imurmurhash@0.1.4, ignore@3.2.7, strip-json-comments@2.0.1, pluralize@1.2.1, globals@9.17.0, estraverse@4.2.0, esquery@1.0.0, esutils@2.0.2, progress@1.1.8, text-table@0.2.0, is-resolvable@1.0.0, user-home@2.0.0, babel-code-frame@6.22.0, levn@0.3.0, optionator@0.8.2, debug@2.6.6, json-stable-stringify@1.0.1, require-uncached@1.0.3, doctrine@2.0.0, chalk@1.1.3, concat-stream@1.6.0, glob@7.1.1, is-my-json-valid@2.16.0, espree@3.4.2, inquirer@0.12.0, js-yaml@3.8.3, file-entry-cache@2.0.0, shelljs@0.7.7, table@3.8.3, escope@3.6.0, lodash@4.17.4)
└── eslint-plugin-mocha@4.9.0 (ramda@0.23.0)
➜  fxa-auth-server git:(master) grunt eslint
Running "eslint:files" (eslint) task
Warning: Failed to load plugin mocha: Cannot find module 'eslint-plugin-mocha'
Referenced from: plugin:fxa/server
Referenced from: /Users/vladikoff/mozilla/fxa-local-dev/fxa-auth-server/.eslintrc Use --force to continue.


➜  fxa-auth-server git:(master) npm i morrme/eslint-plugin-fxa#add-mocha-plugin
npm WARN peerDependencies The peer dependency eslint@^2.0.0 || ^3.0.0 included from eslint-plugin-mocha will no
npm WARN peerDependencies longer be automatically installed to fulfill the peerDependency 
npm WARN peerDependencies in npm 3+. Your application will need to depend on it explicitly.
eslint-plugin-fxa@1.0.0 node_modules/eslint-plugin-fxa
├── eslint@3.19.0 (path-is-inside@1.0.2, natural-compare@1.4.0, strip-bom@3.0.0, imurmurhash@0.1.4, ignore@3.2.7, strip-json-comments@2.0.1, pluralize@1.2.1, globals@9.17.0, estraverse@4.2.0, esquery@1.0.0, esutils@2.0.2, progress@1.1.8, text-table@0.2.0, is-resolvable@1.0.0, user-home@2.0.0, babel-code-frame@6.22.0, levn@0.3.0, optionator@0.8.2, debug@2.6.6, json-stable-stringify@1.0.1, require-uncached@1.0.3, doctrine@2.0.0, chalk@1.1.3, concat-stream@1.6.0, glob@7.1.1, is-my-json-valid@2.16.0, espree@3.4.2, inquirer@0.12.0, js-yaml@3.8.3, file-entry-cache@2.0.0, shelljs@0.7.7, table@3.8.3, escope@3.6.0, lodash@4.17.4)
└── eslint-plugin-mocha@4.9.0 (ramda@0.23.0)
➜  fxa-auth-server git:(master) grunt eslint
Running "eslint:files" (eslint) task
Warning: Failed to load plugin mocha: Cannot find module 'eslint-plugin-mocha'
Referenced from: plugin:fxa/server
Referenced from: /Users/vladikoff/mozilla/fxa-local-dev/fxa-auth-server/.eslintrc Use --force to continue.
```


Does it work for you?

**Pull request closed without merging**